dbt/adapters/spark_cde/cdeapisession.py

```diff
@@ -127,44 +131,51 @@ def execute(self, sql: str, *parameters: Any) -> None:
         adapter_timer.start_timer("submitJob")
         self._cde_connection.submitJob(job_name, job_name, sql_resource, py_resource)
         adapter_timer.end_timer("submitJob")
+        logger.debug("{} .Job created with id: {}".format(job_name, job_name))
+        logger.debug("{} .Job created with sql statement: {}".format(job_name, sql))
         
-        adapter_timer.start_timer("runJob")
+        adapter_timer.start_timer("runJob(spark_exec+sleep)")
         job = self._cde_connection.runJob(job_name).json()
-        self._cde_connection.getJobStatus(job_name)
+        #self._cde_connection.getJobStatus(job_name)
     
         # 3. run the job
         job_status = self._cde_connection.getJobRunStatus(job).json()        
-        # 4. wait for the result       
+        logger.debug("{} .Job status: {}".format(job_name, job_status["status"]))
+        logger.debug("{} .Job run other details: {}".format(job_name, pformat(job_status)))
+
+        # 4. wait for the result    
         while job_status["status"] != CDEApiConnection.JOB_STATUS['succeeded']:
+            logger.debug("{} .Sleeping for {} seconds while polling jobstatus.".format(job_name, DEFAULT_POLL_WAIT))
             time.sleep(DEFAULT_POLL_WAIT)
+            time_slept_while_polling_jobstatus += DEFAULT_POLL_WAIT
             job_status = self._cde_connection.getJobRunStatus(job).json()
```

Comment:
Plan to implement exponential retry and timeout here in future instead of fixed wait and retry.